### DISCUSSION TITLE
Raising the limit of global variables!

### DISCUSSION THREAD

**Discussion by Kiwi1444** [ikemen-engine/Ikemen-GO#2063] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Despite the fact that there is no problems with global limit of helpers on the screen in Ikemen Go and you can overcome limitations in 65 integer variables and 45 float variables by setting varibles inside the helper, this approach has several drawbacks. Three of them cannot be technically fixed due to compatibility issues with Mugen.

1. Helpers have delay relative root for 1 tick. What is essential when we talk about artificial intelligence where every single tick is important for quick reaction

2. All helpers are destroyed at the end of the round. This leads to erasing all values ​​that were assigned to the variables. This is not suitable for long-term storage of information

3. A more complex trigger to access such "external" variables. For example: `Numhelper(helper ID) && helper(helper ID),var(var number)` vs `var(var number)`

4. It is impossible to use the new states: -4 and +1 for "external" variables.

### Describe the solution you'd like

My character uses 6 additional helpers for storage "external" variables. Increasing the variable limit to 500 integer and 500 float will be enough to start with. And in the future we can remove the limit altogether.

### Search terms

variables. variable

**Comment by potsmugen** [comment]
We already have billions of maps.

**Reply by Kiwi1444** [reply]
> We already have billions of maps.

As I understand, map can only return floating values. Would this be a problem for replacing integer variables?

**Reply by potsmugen** [reply]
No. The opposite would be worse.

**Reply by Wftk** [reply]
Maps are good but theres downside we can't define new maps dynamically.

**Comment by Kiwi1444** [comment, marked as answer]
Thanks to [this](https://github.com/ikemen-engine/Ikemen-GO/pull/2409) change, the limit of global variables has been removed.
